manager/controlapi/secret.go

```diff
@@ -157,12 +169,16 @@ func (s *Server) ListSecrets(ctx context.Context, request *api.ListSecretsReques
 //   or if the secret data is too long or contains invalid characters.
 // - Returns an error if the creation fails.
 func (s *Server) CreateSecret(ctx context.Context, request *api.CreateSecretRequest) (*api.CreateSecretResponse, error) {
+	err := s.populateSecretFromPlugin(ctx, request.Spec)
```

Comment:
Thanks @aaronlehmann, the `assignment` solution still means that we need to modify the secret spec payload inside `Assignment_Secret` before we send the assignment (that is, after the secret query in `addTaskDependencies`).  Is this a valid approach?
I think your first concern might cause debugging issues with secret plugins, @diogomonica WDYT?